src/main/js/Widgets/MultipleSelect.js

```diff
@@ -860,16 +878,37 @@ class MultipleSelect {
         .initialize();
 
     // Add the selected options
+    const orderedOptions = [];
     let hasSelectedOptions = false;
     const options = this.element.getOptions();
     for (let i = 0; i < options.length; i++) {
       const option = options[i];
       if (option.isSelected()) {
-        this.selectOption(option);
+        if (this.options.selectedOptions) {
+          orderedOptions.push({
+            opt: option,
+            order: this.options.selectedOptions.indexOf(option.getValue())
+          });
+        } else {
+          this.selectOption(option);
+        }
         hasSelectedOptions = true;
       }
     }
 
+    // Sort the selected options in the order that they were received
+    if (this.options.selectedOptions) {
+      orderedOptions.sort(function(opt1, opt2) {
+        return opt1.order === opt2.order ? 0 :
+            opt1.order > opt2.order ? 1 : -1;
+      });
```

Comment:
If the caller provided us options in order do we need to sort them? Perhaps I need to review the code closer, or we can review it together.